src/orion/core/worker/algo_wrapper.py

```diff
@@ -0,0 +1,191 @@
+        return self.algorithm.is_done
+
+    def score(self, point: Point) -> float:
+        """Allow algorithm to evaluate `point` based on a prediction about
+        this parameter set's performance. Return a subjective measure of expected
+        performance.
+
+        By default, return the same score any parameter (no preference).
+        """
+        return self.algorithm.score(point)
+
+    def judge(self, point: Point, measurements: Any) -> Optional[Dict]:
+        """Inform an algorithm about online `measurements` of a running trial.
+
+        The algorithm can return a dictionary of data which will be provided
+        as a response to the running environment. Default is None response.
+
+        """
+        return self.algorithm.judge(point, measurements)
+
+    @property
+    def should_suspend(self) -> bool:
+        """Allow algorithm to decide whether a particular running trial is still
+        worth to complete its evaluation, based on information provided by the
+        `judge` method.
+
+        """
+        return self.algorithm.should_suspend
+
+    @property
+    def configuration(self) -> Dict:
+        """Return tunable elements of this algorithm in a dictionary form
+        appropriate for saving.
+        """
+        return self.algorithm.configuration
+
+    @contextmanager
+    def warm_start_mode(self):
+        """ Context manager that is used while using points from similar experiments to
+        bootstrap (warm-start) the algorithm.
+
+        The idea behing this is that we don't want the algorithm to modify its state the
+        same way it would if it were observing regular points. For example, the number
+        of "used" trials shouldn't increase, etc.
+
+        New Algorithms or Algo wrappers can implement this method to control how the
+        state of the algo is affected by observing trials from other tasks than the
+        current (target) task.
+        """
+        with self.algorithm.warm_start_mode():
+            yield
+
+    @property
+    def trials_info(self) -> Dict[str, Tuple[Point, Results]]:
+        """ "read-only" property for the dict of points/trials of the wrapped algorithm.
+        """
+        return self._trials_info
+
+    @property
+    def _trials_info(self) -> Dict[str, Tuple[Point, Results]]:
+        """ Proxy for the `_trials_info` field of the wrapped algorithm.
+
+        NOTE: Adding this property just to make sure that if an algo wrapper doesn't
+        override a BaseAlgorithm method (for example when new methods / properties get
+        added to BaseAlgorithm but aren't yet overridden in AlgoWrapper), then things
+        don't accidentally get changed in the Wrapper's `_trials_info` rather than in
+        the wrapped algorithm's.
+        """
+        return self.algorithm._trials_info  # type: ignore
+
+    @_trials_info.setter
+    def _trials_info(self, value: Dict[str, Trial]) -> None:
```

Comment:
What do we assume here for the construction of the algorithm? Algo is built and assigned to self.algorithm, and then the trials_info is created and assigned to `self.unwrapped`? Isn't the `trials_info` created and assigned during the exection of root's algo `__init__`?